### PR TITLE
Time - Quin 

### PR DESCRIPTION
**Pull request by quinqu**
# Task List
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe in your own words what the Model is doing in Rails | A model is the structure of the database your using or a template 
Describe in your own words what the Controller is doing in Rails | A controller is the code/logic that is run when a use interacts with your app, controller methods run the logic behind an action | 
Describe in your own words what the View is doing in Rails | View is the code the browser gets to then display to the human | 
Describe an edge-case controller test you wrote | If a task could not be updated for whatever reason, if not, it should redirect | 
What is the purpose of using strong params? (i.e. the params method in the controller) | So the user can't mess with the ID or other private things about the task. It keeps the task secure | 
How are Rails migrations related to Rails models? | Migrations keep track of and make changes to the model | 
Describe one area of Rails that are still unclear on | Routes is a little confusing sometimes, especially the order of them, however I think I am getting more used to it. |

### CODE REVIEW

File: app/controllers/tasks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+      end 
+
+      @task.destroy
+        
+      redirect_to tasks_path
+      return 
+    end 
+ 
+
+    def mark_complete
```
Comment (beccaelenzil) [inline app/controllers/tasks_controller.rb:81]: Consider adding an action like `mark_incomplete` so that you can change the status of a task.

File: app/views/tasks/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<div class="show"> 
+
+  
+  <div class="show-head">
+  <h2 class='show-header'> Task: <%= @task.name.capitalize()%> </h2>
+  <% if @task.completed_at %>
+  
+    <span class="complete">Completed</span>
```
Comment (beccaelenzil) [inline app/views/tasks/show.html.erb:8]: This badge looks great. Nice styling.

Comment (beccaelenzil) [reply app/views/tasks/show.html.erb:8]: Consider also showing the time it was completed at.

File: app/views/tasks/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+   <% else%>
+
+   <%= link_to "Mark Complete", mark_complete_path(@task), method: :patch %>
```
Comment (beccaelenzil) [inline app/views/tasks/show.html.erb:12]: Minor note: the formatting is a little off here, make sure to tab code under the else.

File: test/controllers/tasks_controller_test.rb [hunk 4]
```diff
@@ -85,40 +79,95 @@
...
+ 
   describe "destroy" do
-    # Your tests go here
+    it "will destroy task" do
+      Task.create(name: "task to delete", description: "DELETE ME")
+
+      task = Task.first
+  
+
+      expect{delete task_path(Task.first.id)}.must_differ "Task.count", -1
```
Comment (beccaelenzil) [inline test/controllers/tasks_controller_test.rb:146]: Remember to include an expectation `must_redirect_to ...`.

Comment (beccaelenzil) [reply test/controllers/tasks_controller_test.rb:146]: Include a test for an invalid_id

File: app/controllers/tasks_controller.rb [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+      @tasks = Task.all
+    end 
+
+    def show 
+      task_id = params[:id]
+      @task = Task.find_by(id: task_id)
+      
+    
+      if @task.nil?
+        flash[:error] = "Could not find task with id: #{task_id}"
```
Comment (beccaelenzil) [inline app/controllers/tasks_controller.rb:14]: woah! working ahead :)

File: app/views/tasks/edit.html.erb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+  <div class="form-form">
+  
+  <h2 class='form-header'> Edit Task </h2>
```
Comment (beccaelenzil) [inline app/views/tasks/edit.html.erb:3]: Consider including this in your partial view by adding to your `locals` hash.

**Comment by beccaelenzil** [comment]
# Task List




## Major Learning Goals/Code Review





| Criteria | yes/no, and optionally any details/lines of code to reference |
| --- | --- |
| At least 6 commits with meaningful commit messages | ✔️
| Routes follow RESTful conventions | ✔️
| Uses named routes (like `_path`) | ✔️
| Creates Models and migrations | ✔️
| Creates styled views | ✔️
| Handles errors like nonexistant tasks | ✔️
| Uses `form_with` to render forms in Rails | ✔️

## Functional Requirements/Manual Testing

| Functional Requirement | yes/no |
| --- | --- |
| Successfully handles index & show | ✔️
| index & show tests pass | ✔️
| Successfully handles: New, Create | ✔️
| New, Create tests pass | ✔️
| Successfully handles: Edit, Update | ✔️
| Successfully handles: Destroy, Task Complete | ✔️

## Overall Feedback

| Overall Feedback | Criteria | yes/no |
| --- | --- | --- |
| Green (Meets/Exceeds Standards) | 5+ in Code Review && 5+ in Functional Requirements | ✔️






## Code Style Bonus Awards



Was the code particularly impressive in code style for any of these reasons (or more...?)

| Quality | Yes? |
| --- | --- |
| Elegant/Clever | ✅
| Descriptive/Readable | ✅
| Concise | ✅
| Logical/Organized | ✅

**Review by beccaelenzil — COMMENTED** [review]
Great work on this project! It is clear that the learning goals for your first rails project were met. Good work incorporating resources, partial views, and strong params. You're off to a great start. Keep up the hard work!
